bitcoin/tx.h

```diff
@@ -77,9 +77,16 @@ int bitcoin_tx_add_output(struct bitcoin_tx *tx, const u8 *script,
 int bitcoin_tx_add_multi_outputs(struct bitcoin_tx *tx,
 				 struct bitcoin_tx_output **outputs);
 
+/* Add a new input to a bitcoin tx.
+ *
+ * For P2WSH inputs, we'll also store the wscript and/or scriptPubkey
+ * Passing in just the {input_wscript}, we'll generate the scriptPubkey for you.
+ * In some cases we may not have the wscript, in which case the scriptPubkey
+ * should be provided. We'll check that it's P2WSH before saving it */
 int bitcoin_tx_add_input(struct bitcoin_tx *tx, const struct bitcoin_txid *txid,
-			 u32 outnum, u32 sequence,
-			 struct amount_sat amount, u8 *script);
+			 u32 outnum, u32 sequence, const u8 *scriptSig,
+			 struct amount_sat amount, const u8 *scriptPubkey,
+			 const u8 *input_wscript);
```

Comment:
I'm sure I will get those 3 `u8[]` params wrong at some point xD

Reply:
Hm yeah a type system might be nice here ;)